tun/tcp_offload_linux.go

```diff
@@ -0,0 +1,615 @@
+package tun
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+	"golang.zx2c4.com/wireguard/conn"
+)
+
+// virtioNetHdrFlags are defined in the kernel in include/uapi/linux/virtio_net.h.
+// These values are used in virtioNetHdr.flags. The kernel symbols begin with
+// VIRTIO_NET_HDR_F_.
+const (
+	virtioNetHeaderFlagsNeedsCSum = 0x1
+)
+
+// virtioNetHdrGSOType are defined in the kernel in include/uapi/linux/virtio_net.h.
+// These values are used in virtioNetHdr.gsoType. The kernel symbols begin with
+// VIRTIO_NET_HDR_GSO_.
+const (
+	virtioNetHeaderGSOTypeNone  = 0x0
+	virtioNetHeaderGSOTypeTCPV4 = 0x1
+	virtioNetHeaderGSOTypeTCPV6 = 0x4
+)
```

Comment:
Should these flags go into x/sys/unix?

Reply:
https://go-review.googlesource.com/c/sys/+/468656

Reply:
Updated in Updated in https://github.com/WireGuard/wireguard-go/pull/64/commits/ba7240c56fef5599358a394511813370be195cfa.